frontend/src/helpers/fashion/index.js

```diff
@@ -0,0 +1,109 @@
+import React from "react";
+import {makeStyles, useTheme} from "@material-ui/core/styles";
+import clsx from "clsx";
+
+/***
+ * Style a component similar to {@link https://material-ui.com/styles/basics/#styled-components-api|material-ui styled-components}
+ * which was inspired by {@link https://styled-components.com/|styled-components}
+ *
+ * This is the similar to {@link fashion} but you modify any single css class of your liking simillar to how you would do it with
+ * {@link makeStyles}. This function might be useful in situations where you want to override material-ui styling.
+ *
+ * @example Fashion with access to the theme.
+ * const RoundedButton = hardFashion(Button, (theme) => ({
+ *     root: {
+ *         backgroundColor: "#4F7CE0",
+ *         borderRadius: 15,
+ *         "&:hover": {
+ *             backgroundColor: "#7290D3"
+ *         }
+ *         padding: theme.padding(2)
+ *     },
+ *     label: {
+ *         color: "white"
+ *     }
+ * }));
+ * @example A shorthand for making styles simply with an object.
+ * const RoundedButton = hardFashion(Button, {
+ *     root: {
+ *         backgroundColor: "#4F7CE0",
+ *         borderRadius: 15,
+ *         "&:hover": {
+ *             backgroundColor: "#7290D3"
+ *         }
+ *     },
+ *     label: {
+ *         color: "white"
+ *     }
+ * });
+ *
+ * @typedef {Object} Style
+ * @typedef {Object} Theme
+ *
+ * @param {React.ComponentClass} BaseComponent The component to style
+ * @param {function(Theme, React.ComponentProps) | Style} styling Can either be an object with styles or a function(theme, props) that returns the styles
+ * @returns {React.FunctionComponent} A new React Component with the applied styles
+ */
+export function hardFashion(BaseComponent, styling) {
+    return function FashionedComponent(props) {
+        const theme = useTheme();
+        let useStyles;
+
+        if (typeof styling == "function") {
+            const styles = styling(theme, props);
+            useStyles = makeStyles(styles);
+        } else if (typeof styling == "object") {
+            useStyles = makeStyles(styling);
+        } else {
+            throw new Error(`Cannot fashion with ${styling}`)
+        }
+
+        const classes = useStyles();
+        return (<BaseComponent classes={{...classes, ...props.classes}} {...props}/>);
+    };
+}
```

Comment:
I like this concept, but could we make the function names more descriptive `fashion` and `hardFashion` are a bit to abstract. If we aren't going to use the styled-components why not call these functions something like `wrapWithStyle` and returns a `StyledComponent`.

Reply:
_Wrapping a component with style_ might be more intuitive but then again the idea of _to fashion a component_, a synonym for _to style_ can be pretty intuitive as well. At this point, I think the naming can be quite subjective and makes a minor difference. The only thing that is not as intuitive is the idea of doing it _hard_ which means to override all styles completely. Then again, the by design, the people who just want to style can use the simple `fashion` function. The ones that want to go deep and override material-ui styles can go with `hardFashion`.

For reference, material-ui has this kind of styled-component api, however I can't seem to find a way to access the theme using their API. The custom one I made does that by utilizing react hooks.